dataset.py

```diff
@@ -431,7 +434,9 @@ def create_fraudulent_samples(real_samples_data: pd.DataFrame, mean_isoscapes: l
   count = 0
 
   for coord, lab_samp in real_samples_code:
-    if lab_samp.size <= 1 :
+    if count >= fake_amount:
```

Comment:
if the desired fraud percent is > 50% (lets say 90% fraud), then the number of fraud items will exceed the number of real samples.  If 90% fraud is chosen, then the fraud count will be 10x the real sample count.

So I think this loop may have to run differently -- iterating on the desired count of fake rows and choosing a real sample by random as well as a random location.